### PR TITLE
updated catalog modal design

### PR DESCRIPTION
**Pull request by thisiskaransgit**
**Description**
Updating Modal design

**Notes for Reviewers**
1. Incorporating feedback that was overlooked in the previous update #1349 
2. Reason for Delay: A particular modal had a design bug visible, it turns out the source file had a syntax error, which I fixed.
3. Fixing responsiveness issues

**[Signed commits](https://docs.meshery.io/project/contributing#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

File: _includes/modal.html [hunk 1]
```diff
@@ -1,48 +1,39 @@
...
       </div>
-
-      <div>
-        {%if pattern.patternInfo%}
+      <div class="modal-info">
+        <h2 class="modal-h2">{{pattern.name | upcase }} {% include read-more.html %}</h2>
+        {% if pattern.patternInfo %}
         <h3 class="modal-h3">What this pattern does</h3>
-        <p>{{pattern.patternInfo}}</p>
+        <p>{{pattern.patternInfo | replace: '\\n', '<br/>'| replace: '\n', '<br/>' }}</p>
```
Comment (thisiskaransgit) [inline _includes/modal.html:24]:
```suggestion
        <p>{{pattern.patternInfo}}</p>
```

File: _includes/partials/individual-component.html [hunk 2]
```diff
@@ -14,7 +14,7 @@ <h4 class="related-patterns">Related Patterns</h4>
   {%if pattern.patternInfo%}
   <div>
     <h3>What this pattern does:</h3>
-    <p>{{pattern.patternInfo}}</p>
+    <p>{{pattern.patternInfo | replace: '\\n', '<br/>'| replace: '\n', '<br/>' }}</p>
```
Comment (thisiskaransgit) [inline _includes/partials/individual-component.html:17]:
```suggestion
    <p>{{pattern.patternInfo}}</p>
```

**Review by Chadha93 — CHANGES_REQUESTED** [review]
@thisiskaransgit Alignment and padding issues in the header. Also, let's get clear on the placement of x, it's odd to have it on the left, I'm aware of the design that we have a badge on the top-right. 

Just to confirm, did we ever had a complete UX and design made for this modal? cc @ritiksaxena124 
image

**Comment by thisiskaransgit** [comment]
> @thisiskaransgit Alignment and padding issues in the header. Also, let's get clear on the placement of x, it's odd to have it on the left, I'm aware of the design that we have a badge on the top-right.
> 
> Just to confirm, did we ever had a complete UX and design made for this modal? cc @ritiksaxena124 image

@Chadha93 No major design was needed here, and the placement of elements on the position was intentional. The suggestion was provided in Slack, since the last PR wasn't concluded with all the requests, this one was needed. 

<summary>
<details>

1. note the need for a different color for the “X”(previously it was black)
2. You might add a bit of space between the design diagram and the buttons or put the buttons at the bottom of the modal.
3. or put an outline around the diagram’s screenshot (which are all the same right now, but will become MeshMap Snapshots)
4. Note the disjointed type label in the Related Patterns card (something to fix (bring inline with the rest of the design)). (edited) 
5. Do not use bold text.
6. Change the color of the external link button, too.
7. give it some space between the card name and make is the larger to be almost the same size as the card name.
8. The white space between the card name and the first “WHAT THIS PATTERN….” is wasted and can be cut-down.
9. The “Related Patterns” section title can be captialized.
10. Finally, if you move the screenshot, buttons and related patterns back to the left side, where they were, you can vertically split the card into two sections such that the left side has a different background than the main card, providing clarity to the user, making it easier to digest the information. (edited) 
11. It's fine to keep the card name in bold.

</details>
</summary>

**Comment by ritiksaxena124** [comment]
Suggestion: we should show only relevant details on modal not all details because what is the point of having a separate page when we are showing all details on modal

**Review by Chadha93 — APPROVED** [review]

**Pull request merged**